### PR TITLE
Ellipse marker symbol layer fixes and code improvements

### PR DESCRIPTION
**Pull request by nirvn**
## Description

Giving a bit of :heart: to this under-maintained marker symbol layer. 

The chunk of the PR upgrades the ellipse marker symbol layer away from using strings into using an enum to identify shape type. This does us a couple of favors: i) it's faster code, ii) it allows us to use switch( shape ) and be warned of missing code addition when adding new shapes, and iii) it harmonizes code styling with that of the simple marker symbol layer, enabling easier porting of feature from one marker type to the other.

### CODE REVIEW

**Comment by nirvn** [comment]
TIL: our ellipse marker symbol layer failed to draw a selected state for non-filled shapes (cross)

**Comment by nirvn** [comment]
@nyalldawson , this is ready to review -- test failures unrelated.

File: src/core/symbology/qgsellipsesymbollayer.h [hunk 1]
```diff
@@ -49,16 +75,56 @@ class CORE_EXPORT QgsEllipseSymbolLayer: public QgsMarkerSymbolLayer
...
 
-    void setSymbolName( const QString &name ) { mSymbolName = name; }
-    QString symbolName() const { return mSymbolName; }
+    /**
+     * Sets the rendered ellipse marker shape using a symbol \a name.
+     * \see setShape()
+     * \see shape()
+     * \deprecated since QGIS 3.20
+     */
+    Q_DECL_DEPRECATED void setSymbolName( const QString &name ) { mShape = decodeShape( name ); }
```
Comment (nyalldawson) [inline src/core/symbology/qgsellipsesymbollayer.h:84]: These need SIP_DEPRECATED too

**Comment by nirvn** [comment]
@nyalldawson , done.

**Pull request merged**